sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -531,15 +536,13 @@ export class CryptographyClient {
       }
     }
 
-    let providers = this.providers.filter((p) => p.supportsOperation(operation));
-    if (algorithm) {
-      providers = providers.filter((p) => p.supportsAlgorithm(algorithm));
-    }
+    const providers = this.providers.filter((p) => p.isSupported(algorithm, operation));
```

Comment:
Per the comment above, where is the change to *not* add the remote provider for local-only operation, i.e. when called with the JWK-only constructor?

Reply:
The diff hides it, but we make that determination when we first build the providers list (which is cached after the first call)